scripts/google_replicate.py

```diff
@@ -0,0 +1,240 @@
+import base64
+import requests
+import urllib
+
+from google.cloud import storage
+from google.cloud.storage import Blob
+from google_resumable_upload import GCSObjectStreamUpload
+from google.auth.transport.requests import AuthorizedSession
+
+import logging as logger
+from indexclient.client import IndexClient
+
+from errors import APIError
+from settings import PROJECT_MAP, INDEXD, GDC_TOKEN
+from utils import get_bucket_name
+
+indexclient = IndexClient(
+    INDEXD["host"],
+    INDEXD["version"],
+    (INDEXD["auth"]["username"], INDEXD["auth"]["password"]),
+)
+
+DATA_ENDPT = "https://api.gdc.cancer.gov/data/"
+DEFAULT_CHUNK_SIZE_DOWNLOAD = 2048000
+DEFAULT_CHUNK_SIZE_UPLOAD = 1024 * 20 * 1024
+
+
+class DataFlowLog(object):
+    def __init__(self, copy_success=False, index_success=False, message=""):
+        self.copy_success = copy_success
+        self.index_success = index_success
+        self.message = message
+
+
+def bucket_exists(bucket_name):
+    """
+    check if bucket_name exists or not!
+    """
+    client = storage.Client()
+    bucket = client.bucket(bucket_name)
+    return bucket.exists()
+
+
+def blob_exists(bucket_name, blob_name):
+    """
+    check if blob exists or not!
+    """
+    if bucket_exists(bucket_name):
+        client = storage.Client()
+        bucket = client.bucket(bucket_name)
+        blob = Blob(blob_name, bucket)
+        return blob.exists()
+    return False
+
+
+def check_blob_name_exists_and_match_md5_size(sess, bucket_name, blob_name, fi):
+    """
+    check if blob object exists or not
+    Args:
+        bucket_name(str): the name of bucket
+        blob_name(str): the name of blob
+        fi(dict): file info dictionary
+    Returns:
+        bool: indicating value if the blob is exist or not
+    """
+    url = "https://www.googleapis.com/storage/v1/b/{}/o/{}".format(
```

Comment:
can you not use the `storage` library to get this information? I know google's api's are sometimes weird, so it's entirely possible you have to use the REST API, I'm just curious